server/db/driver/pg/accounts.go

```diff
@@ -53,12 +54,18 @@ func (a *Archiver) Accounts() ([]*db.Account, error) {
 	}
 	defer rows.Close()
 	var accts []*db.Account
+	var accountID, pubkey, feeCoin []byte
+	var brokenRule byte
 	for rows.Next() {
 		a := new(db.Account)
-		err = rows.Scan(&a.AccountID, &a.Pubkey, &a.FeeAddress, &a.FeeCoin, &a.BrokenRule)
+		err = rows.Scan(&accountID, &pubkey, &a.FeeAddress, &feeCoin, &brokenRule)
 		if err != nil {
 			return nil, err
 		}
+		copy(a.AccountID[:], accountID)
+		a.Pubkey = dex.Bytes(pubkey)
+		a.FeeCoin = dex.Bytes(feeCoin)
+		a.BrokenRule = account.Rule(brokenRule)
```

Comment:
Whoops:
`failed to retrieve accounts: sql: Scan error on column index 3, name "fee_coin": unsupported Scan, storing driver.Value type <nil> into type *dex.Bytes`
I suppose it is because sql doesn't work well with custom types? Have removed all custom types from the `Scan` just in case.

Reply:
Custom types need either (1) built-in underlying type, or (2) define Scan and Value methods to satisfy sql.Scanner and sql.Valuer interfaces.  We have those define for many type.  I suggest looking at those for examples. e.g. for `AccountID`: https://github.com/decred/dcrdex/blob/95d8c3ffd08d6c9c87d7033b25b73a6c14f03027/server/account/account.go#L41-L59, which does the copy that you are doing manually now.

However, the error here indicates that `dex.Bytes`, which is just `[]byte` doesn't handle NULL values in the table.  Note that the NULL issue is so common that there are standard types like `sql.NullString` and `sql.NullInt64` for this reason.  For byte slices, I think it should just work with NULLs.  How do you get this error (what code and test)?

Reply:
I can repro the error by changing `getAccount` to use `dex.Bytes` instead of `[]byte`.  I'm OK with Scanning into a `[]byte` and just converting like you have.  But we can definitely scan into an account.AccountID directly because we have implemented sql.Scanner for that.

Reply:
Adding a `Scan` method to `dex.Bytes` looks to has solved the problem. Is this alright? It seems odd that this is needed since scanning when the value is not null used the underlying type's Scan.